### PR TITLE
opcode: add RecvMsgMulti

### PR DESCRIPTION
**Pull request by FrankReh**
Adds the RecvMsgMulti opcode.

Updated the unit test to use it.

Also fixes the RecvMsg handling of the seq.flags field. It wouldn't have handled a fixed file descriptor.

Also improves the documentation to RecvMsg, to better describe the buf_group parameter and the *Multi* aspect of the operation.

### CODE REVIEW

**Comment by FrankReh** [comment]
I used some spacing in the documentation for the parameters. I don't know how that will look when it is actually rendered.

File: src/opcode.rs [hunk 1]
```diff
@@ -509,6 +509,58 @@ opcode!(
...
+    /// of the rest of the buffer in combination with the initial msghdr structure submitted with
+    /// the request. Use [`types::RecvMsgOut`] to parse the data received and access its
+    /// components.
+    ///
+    /// The recvmsg multishot variant is available since kernel 6.0.
+
+    #[derive(Debug)]
+    pub struct RecvMsgMulti {
+        fd: { impl sealed::UseFixed },
+        msg: { *mut libc::msghdr },
```
Comment (lucab) [inline src/opcode.rs:539]: I think this doesn't strictly need to be `mut`.

Comment (FrankReh) [reply src/opcode.rs]: Done. Thanks.

File: src/opcode.rs [hunk 2]
```diff
@@ -509,6 +509,58 @@ opcode!(
...
+    /// should issue a new request.
+    ///
+    /// Unlike [`RecvMsg`], this multishot recvmsg will prepend a struct which describes the layout
+    /// of the rest of the buffer in combination with the initial msghdr structure submitted with
+    /// the request. Use [`types::RecvMsgOut`] to parse the data received and access its
+    /// components.
+    ///
+    /// The recvmsg multishot variant is available since kernel 6.0.
+
+    #[derive(Debug)]
```
Comment (quininer) [inline src/opcode.rs:536]: This should remove?

Comment (quininer) [reply src/opcode.rs:536]: It looks like some opcode is not implemented Debug.

**Pull request merged**